python/lsst/daf/butler/core/storageClass.py

```diff
@@ -63,8 +69,8 @@ def __init__(self, name, pytype=None, components=None, assembler=None):
         else:
             # We set a default assembler so that a class is guaranteed to
             # support something.
-            self._assemblerClassName = None
-            self._assembler = CompositeAssembler
+            self._assembler = self.defaultAssembler
+            self._assemblerClassName = self.defaultAssemblerName
```

Comment:
Is this necessary if the StorageClass does not have any components?

Reply:
with a tweak to butler.py I can make it so that I only default when components are present. I'll make that change and see what happens.